pandas/io/formats/format.py

```diff
@@ -730,6 +731,11 @@ def to_string(self) -> None:
         """
         Render a DataFrame to a console-friendly tabular output.
         """
+        # Note: the to_string method only accepts IO whereas to_html and
```

Comment:
Thanks for the notes - are these bugs?

Reply:
These are the sort of inconsistencies between to_html, to_string and to_latex that I was hoping to find by adding type hints.

This should help with refactoring the three methods to share buffer handling code.

Can't really call them bugs when the documentation doesn't say that filenames are accepted (even though they are for to_html and to_latex).
```
buf : StringIO-like, optional
Buffer to write to.
```

Reply:
Do you know what the lifecycle of `self.buf = cast(...)` is? Would that cast still just stay local to the function?